lib/teiserver/game/servers/balancer_server.ex

```diff
@@ -11,7 +11,9 @@ defmodule Teiserver.Game.BalancerServer do
   @algos_allowing_fuzz ~w(loser_picks force_party)
   @spec start_link(List.t()) :: :ignore | {:error, any} | {:ok, pid}
   def start_link(opts) do
-    GenServer.start_link(__MODULE__, opts[:data], [])
+    # https://www.erlang.org/docs/23/man/erlang#process_flag_max_heap_size
+    max_heap_size = 50 * 1024 * 1024 / 8
+    GenServer.start_link(__MODULE__, opts[:data], max_heap_size: max_heap_size)
```

Comment:
the dialyzer warnings show that this is incorrect, there is no such argument `max_heap_size` for `start_link/3`. The doc you linked shows that it is another function entirely: `Process.flag/2` (the [elixir version](https://hexdocs.pm/elixir/Process.html#flag/2) points to the erlang one since it's the same).